specification/app/resource-manager/Microsoft.App/preview/2023-11-02-preview/JavaComponents.json

```diff
@@ -0,0 +1,481 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/JavaComponent"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Java Component": {
+            "$ref": "./examples/JavaComponents_Get.json"
+          },
+          "Get Java Component with ServiceBinds": {
+            "$ref": "./examples/JavaComponents_Get_ServiceBind.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "JavaComponents"
+        ],
+        "summary": "Creates or updates a Java Component.",
+        "description": "Creates or updates a Java Component in a Managed Environment.",
+        "operationId": "JavaComponents_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "environmentName",
+            "in": "path",
+            "description": "Name of the Managed Environment.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the Java Component.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "javaComponentEnvelope",
+            "in": "body",
+            "description": "Configuration details of the Java Component.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/JavaComponent"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "No change performed on the Java Component.",
+            "schema": {
+              "$ref": "#/definitions/JavaComponent"
+            }
+          },
+          "201": {
+            "description": "Java Component create or update has been started.",
+            "schema": {
+              "$ref": "#/definitions/JavaComponent"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
```

Comment:
Consider use `../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse`, please also update other places

Reply:
Hi @ms-henglu thanks for the suggestion, may I follow the pattern of other files and use this "DefaultErrorResponse" for now, we can have a work item for this so that we can update this part later together